### PR TITLE
chore: update dependencies list

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Fixes https://github.com/appsmithorg/appsmith/issues/18534
Depends on https://github.com/appsmithorg/design-system/pull/261, https://github.com/appsmithorg/appsmith/pull/18231

## Type of change

Chore

## How Has This Been Tested?
- Jest
- Cypress

### Test Plan
1. General regression 
2. Let us test the `CopyToClipboard` component as a sample to confirm that the moved libraries haven't broken functionality. https://docs.google.com/spreadsheets/d/1np7jQdiQa0nyryOBnNa927NkGDplG9M2gb7qnoZIIyM/edit#gid=2010105672

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/yarn.lock [hunk 1]
```diff
@@ -12931,13 +12899,6 @@ react-router@5.2.0, react-router@^5.1.2:
     tiny-invariant "^1.0.2"
     tiny-warning "^1.0.0"
 
-react-router@6.3.0:
```
Comment (albinAppsmith) [inline app/client/yarn.lock:12934]: Why is react router deleted from here?

File: app/client/yarn.lock [hunk 2]
```diff
@@ -12908,14 +12884,6 @@ react-router-dom@^5.1.2:
     tiny-invariant "^1.0.2"
     tiny-warning "^1.0.0"
 
-react-router-dom@^6.3.0:
```
Comment (albinAppsmith) [inline app/client/yarn.lock:12911]: Why this is removed?

**Review by albinAppsmith — APPROVED** [review]

File: app/client/package.json [hunk 3]
```diff
@@ -45,7 +45,7 @@
     "cypress-log-to-output": "^1.1.2",
     "dayjs": "^1.10.6",
     "deep-diff": "^1.0.2",
-    "design-system": "npm:@appsmithorg/design-system@1.0.34",
+    "design-system": "npm:@appsmithorg/design-system@1.0.34-alpha.4",
```
Comment (yaldram) [inline app/client/package.json:48]: Make sure you don't use the alpha version

**Review by yaldram — APPROVED** [review]

File: app/client/package.json (hunk 3 above)
Comment (tanvibhakta) [reply app/client/package.json:48]: This PR needs to be approved and tested before I can merge the corresponding design-system PR to the design-system release branch and update the release number :) For testing purposes, we always use alphas.

**Comment by shadabbuchh** [comment]
Tested this PR. The `Copy to Clipboard` button wrks fine.

**Pull request merged**